### PR TITLE
Add Super+Scroll actions

### PR DESCRIPTION
**Pull request by lenemter**
Fixes https://github.com/elementary/gala/issues/815
Fixes https://github.com/elementary/gala/issues/1178
Closes https://github.com/elementary/switchboard-plug-mouse-touchpad/issues/215

### CODE REVIEW

File: src/WindowManager.vala [hunk 1]
```diff
@@ -157,6 +157,26 @@ namespace Gala {
                 });
             }
 #endif
+
+            var scroll_action = new SuperScrollAction (display);
+            scroll_action.triggered.connect (handle_super_scroll);
+
+            stage.add_action_full ("super-scroll-action", CAPTURE, scroll_action);
```
Comment (leolost2605) [inline src/WindowManager.vala:164]: Currently all of the stage stuff and keybindings etc. is done in show_stage maybe we should do this there as well?

**Review by leolost2605 — CHANGES_REQUESTED** [review]
I really really love this!

**Review by leolost2605 — COMMENTED** [review]
On second thought I think it might be better to make the `void triggered ()` signal a `bool triggered ()`, return that for propagation and have everything that needs a super scroll action add their own. This would stay more in line with how events, etc. are generally handled and avoids having to add even more api to the windowmanager.
What do you think?

**Comment by leolost2605** [comment]
> @leolost2605 we need to implement our own objects hierarchy for this to make sense

I'm not sure I understand what you mean here?

**Comment by lenemter** [comment]
How are we gonna propagate the signal in this case? It's not an event, so we can't use Clutter... TBH I don't understand exactly what you mean in your original comment.

**Comment by leolost2605** [comment]
I meant something like the last commit in #2182 does

**Pull request closed without merging**